src/web/PopupContainer.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+* PopupContainer.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Common parent of all components rendered into a popup. Calls onShow and onHide
+* callbacks when the popup is hidden (i.e. "closed" but still rendered as hidden)
+* and re-shown.
+*/
+
+import React = require('react');
+import PropTypes = require('prop-types');
+import FocusManager from './utils/FocusManager';
+
+export interface PopupContainerProps {
+    style: React.CSSProperties;
+    onMouseEnter?: (e: any) => void;
+    onMouseLeave?: (e: any) => void;
+    hidden?: boolean;
+}
+
+export interface PopupContainerContext {
+    focusManager?: FocusManager;
+}
+
+export interface PopupComponent {
+    onShow: () => void;
+    onHide: () => void;
+}
+
+export class PopupContainer extends React.Component<PopupContainerProps, {}> {
+    static contextTypes: React.ValidationMap<any> = {
+        focusManager: PropTypes.object
+    };
+    static childContextTypes: React.ValidationMap<any> = {
+        focusManager: PropTypes.object,
+        popupContainer: PropTypes.object
+    };
+
+    private _popupComponentStack: PopupComponent[] = [];
+
+    constructor(props: PopupContainerProps, context: PopupContainerContext) {
+        super(props, context);
+    }
+
+    getChildContext() {
+        return {
+            focusManager: this.context.focusManager,
+            popupContainer: this as PopupContainer
+        };
+    }
+
+    render() {
+        let style = this.props.style;
+        if (this.props.hidden) {
+            style['visibility'] = 'hidden';
```

Comment:
You're modifying the object passed into props by the caller. You should clone it before adding to it. Or, better yet, pass in an array of styles to the div below. That way, you can avoid cloning it.

Reply:
Fixed by cloning. Div doesn't seem to take style arrays.

```
interface HTMLAttributes<T> extends DOMAttributes<T> {
        ...
        style?: CSSProperties;
        ...
```